protocol/testing/e2e/funding/funding_e2e_test.go

```diff
@@ -141,11 +141,15 @@ func TestFunding(t *testing.T) {
 
 	for name, tc := range tests {
 		t.Run(name, func(t *testing.T) {
-			tApp := testapp.NewTestAppBuilder(t).WithGenesisDocFn(func() (genesis types.GenesisDoc) {
-				genesis = testapp.DefaultGenesis()
-				genesis.GenesisTime = GenesisTime
-				return genesis
-			}).Build()
+			tApp := testapp.NewTestAppBuilder(t).
+				// UpdateIndexPrice only contacts the tApp.App.Server causing non-determinism in the
+				// other App instances in TestApp used for non-determinism checking.
+				WithNonDeterminismChecksEnabled(false).
```

Comment:
For e2e tests that require setting up in-memory state, instead of disabling non-determinism checks altogether (zero coverage), would it make sense to apply the required in-memory setup to the 3 new instances (at least some coverage)? 

For instance, allow test cases to pass in `func SetupApp(app *app.App)` so changes like `pricefeed_testutil.UpdateIndexPrice(` can be applied to not just `tApp.App` but the 3 other instances. 

In the case of the funding e2e test, the better way is probably to overwrite `Prepare/ProcessProposal` to accept the new oracle price, but unsure if other tests could benefit from above.

Reply:
For things that mutate keepers, we should rely on changing the tests to use genesis/CheckTx instead.

For queries, we could expose new capabilities like how TestApp.CheckTx is invoked, we could have something like TestApp.ServerMsg that invokes it on all the instances but I'm pretty undecided as to what to do here as this change moves the needle pretty far.

Reply:
> For things that mutate keepers, we should rely on changing the tests to use genesis/CheckTx instead.

For tests where oracle prices need to be be updated more than once during the lifespan, `genesis` isn't sufficient. I guess in this case we should use `DeliverTxsOverride` to force a `MsgUpdatePrice` transaction?

Reply:
We could ensure that they source the same pricing information.